app/javascript/src/utils/codemirror/indent.js

```diff
@@ -141,31 +140,28 @@ export function indentMultilineInserts({ state, dispatch }, transaction) {
   // Only perform this function if transaction is of an expected type performed by the user to prevent infinite loops on changes made by CodeMirror
   if (transaction.transactions.every(tr => ["input.paste", "input.complete"].includes(tr.annotation(Transaction.userEvent)))) {
     const [range] = transaction.changedRanges
-    const rangeLine = state.doc.lineAt(range.fromB)
-    const text = transaction.state.doc.toString().slice(range.fromB, range.toB)
+    let text = transaction.state.doc.toString().slice(range.fromB, range.toB)
     const splitText = text.split("\n")
 
-    let startIndentCount = 0
-    let firstIndentCount = 0
-    const mappedText = splitText.map((line, i) => {
-      if (!i) {
-        firstIndentCount = getIndentCountForText(line)
-        startIndentCount = getIndentCountForText(rangeLine.text) - firstIndentCount
-
-        return line.replace(/^\s+/, "")
+    let indentDifference = 0
+    //Dont touch single-line content (allow pasting of whitespace characters without treating them as indents)
+    if(splitText.length >= 2){
```

Comment:
I think ultimately while this helps, it still points towards larger issues. When I wrote this it was originally only meant for autocomplete, and fixing some pasting business was just a nice bonus, but it's proven to be less than helpful. Can I propose we just disable it for pasting and only keep it for autocomplete?

Reply:
yep i see your point! its pretty clear that the pasting is janky because of inconsistencies elsewhere. and im not a fan of whitespace characters being treated as indents (and possibly getting removed). 
might be best to use the code as it currently is live for the autocomplete, and then make only a minor adjustment to the paste without changing what is actually getting pasted. i still want to deal with those wrong indents on the first line though. just leave the PR laying around for now and i will have a look at it tomorrow 😀

Reply:
CodeMirror already did a decent job of automatically indenting pastes, my attempts to make it better made it worse :D

Removing it should be as simple as removing `"input.paste"` from https://github.com/Mitcheljager/workshop.codes/pull/417/files#diff-57562ceac603867d58c8cbfb32b690dde45b10366da68ac14eb56f19b422b8a5R141

Reply:
Alright lets just take the most straightforward and simplest solution then! But i still want to have a look at that first indent at one point or another :P

Reply:
Sure!